pyccel/ast/numpyext.py

```diff
@@ -881,30 +882,59 @@ def fprint(self, printer, lhs):
         return alloc +'\n' + stmt
 
 #==============================================================================
-class Rand(Real):
+class Rand(Function, PyccelAstNode):
 
     """
       Represents a call to  numpy.random.random or numpy.random.rand for code generation.
 
     """
-    def __new__(cls, arg = Nil()):
-        return Real.__new__(cls, arg)
+    _dtype = 'real'
+    _precision = default_precision['real']
 
-    def __init__(self):
-        Real.__init__(self, None)
+    def __init__(self, *args):
+        if len(args) == 0:
+            self._shape = ()
+        else:
+            self._shape = args
 
-    @property
-    def arg(self):
-        return self._args[0]
+        self._rank  = len(self.shape)
+
+        assumptions = {'real':True}
+        ass_copy = assumptions.copy()
+        self._assumptions = StdFactKB(assumptions)
+        self._assumptions._generator = ass_copy
 
     @property
-    def rank(self):
-        return 0
+    def order(self):
+        return 'C'
 
-    def fprint(self, printer):
+    def fprint(self, printer, lhs, stack_array=False):
         """Fortran print."""
 
-        return 'rand()'
+        lhs_code = printer(lhs)
+        stmts = []
+
+        if self.rank>0:
+            # Create statement for allocation
+            if not stack_array:
+                # Transpose indices because of Fortran column-major ordering
+                shape = self.shape[::-1]
+
+                if isinstance(self.shape, (PythonTuple,Tuple,tuple)):
+                    # this is a correction. problem on LRZ
```

Comment:
What problem do we have on LRZ?

Reply:
I don't know. The allocation code was copied and pasted from the Full class (this should probably be done once. Maybe in the allocation super class that we could create for issue #307)
If I work my way back up the tree it seems that @saidctb wrote this comment originally

Reply:
@saidctb Do you remember what this problem was about? Can we just remove those lines?

Reply:
@ratnania added this part so I'm not sure

Reply:
After taking another look at it, I think that we never enter the code at line 927 just below. In fact `shape` should never be a scalar, and the comment `# this is a correction. problem on LRZ` is probably a leftover. So I suggest we replace the whole `if/else` block with just
```
shape_code = ', '.join('0:' + printer(Add(i, Integer(-1))) for i in shape)
```
The same change should be made to the `Full` class, and maybe to other places too.

Reply:
I think you are right. I tried to ensure that shape is always a tuple, so it shouldn't be used. And codacy's coverage seems to indicate that it isn't used